plugins/src/test/java/org/smartregister/fhir/gateway/plugins/SyncAccessDecisionTest.java

```diff
@@ -997,6 +1000,76 @@ public void testPostProcessWithListModeHeaderSearchByTagPaginateEntriesBundle()
                 resultContent);
     }
 
+    @Test
+    public void testPostProcessCallsUpdateLocationLineage() throws IOException {
+        testInstance =
+                Mockito.spy(createSyncAccessDecisionTestInstance(Constants.SyncStrategy.LOCATION));
+        FhirContext fhirR4Context = mock(FhirContext.class);
+        IGenericClient iGenericClient = mock(IGenericClient.class);
+
+        testInstance.setFhirR4Context(fhirR4Context);
+        testInstance.setFhirR4Client(iGenericClient);
+
+        RequestDetailsReader requestDetailsSpy = Mockito.mock(RequestDetailsReader.class);
+        RequestTypeEnum requestTypeEnumMock = Mockito.mock(RequestTypeEnum.class);
+
+        Mockito.when(requestDetailsSpy.getRequestPath()).thenReturn("Location/123");
+        Mockito.when(requestDetailsSpy.getRequestType()).thenReturn(requestTypeEnumMock);
+        Mockito.when(requestTypeEnumMock.name()).thenReturn("POST");
+        Mockito.when(requestDetailsSpy.getResourceName())
+                .thenReturn(Constants.SyncStrategy.LOCATION);
+
+        Location location = new Location();
+        location.setId("Location/123");
+
+        String responseJson = FhirContext.forR4().newJsonParser().encodeResourceToString(location);
+        HttpResponse fhirResponseMock =
+                Mockito.mock(HttpResponse.class, Answers.RETURNS_DEEP_STUBS);
+        TestUtil.setUpFhirResponseMock(fhirResponseMock, responseJson);
+        Mockito.doReturn("123")
+                .when(testInstance)
+                .getLocationId(Mockito.anyString(), Mockito.any());
+
+        try (MockedStatic<LocationHelper> locationHelperMock =
+                Mockito.mockStatic(LocationHelper.class)) {
+            locationHelperMock
+                    .when(() -> LocationHelper.updateLocationLineage(any(), any()))
+                    .thenReturn(location);
+
+            testInstance.postProcess(requestDetailsSpy, fhirResponseMock);
+            locationHelperMock.verify(
+                    () -> LocationHelper.updateLocationLineage(eq(iGenericClient), eq("123")));
+
+            Assert.assertNotNull(location);
+            Assert.assertEquals("Location/123", location.getId());
```

Comment:
These asserts are redundant for the unit tests given the values are being set by Line 1022 and 1023. We should remove them.